Introduce QT GUI

VanDavv opened the pull request:
This PR introduces QT-based GUI

![Screenshot 2021-11-03 at 11 52 13](https://user-images.githubusercontent.com/5244214/140047846-c3b3ab2c-f629-486f-bf26-1a953f680650.png)

You can see a "play-around" video here - https://www.youtube.com/watch?v=sCZpsFQBffk

Some of the options you can see disabled, as changing these requires demo script restart - which will be added (in todos), either in this PR or in a separate one if we'd like to split these

Together with this PR, the demo code itself got refactored - it's now class-based with the main class called `Demo`. This will allow us to import and modify the demo code or specify callbacks easier - but I didn't put too much into this aspect yet as it was not the core aspect of this PR
The change was required since the demo script now runs in a separate thread and shares its memory with main thread.

TODO:
- Tested on Mac, will need to test it on Ubuntu and Windows too before merge
- Investigate FPS limitations (it's pretty poor on video, have to check if it's due to USB speed or QT Signals speed)
- letterboxing instead of frame resize
- enable options that require demo restart

szabi-luxonis reviewed (commented):
Tested w/ python 3.6 and 3.10 on ubuntu 21.10, works fine apart from some issues:

Rectified left/right frames are flipped relative to depth frames.

Seems to be placement issues here:
![image](https://user-images.githubusercontent.com/60790666/140336486-4d94913a-b086-4577-9249-324c17cdf250.png)

VanDavv commented:
@szabi-luxonis addressed the issues - I didn't see the issue on my side with the placement, but moved these values a bit lower just in case.
![Screenshot 2021-11-08 at 10 18 11](https://user-images.githubusercontent.com/5244214/140715679-6df16375-e065-4cfd-b256-8b899ea779bc.png)
